tariff/entsoe.go

```diff
@@ -0,0 +1,152 @@
+package tariff
+
+import (
+	"errors"
+	"net/http"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/tariff/entsoe"
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/request"
+	"github.com/evcc-io/evcc/util/transport"
+)
+
+type Entsoe struct {
+	*request.Helper
+	*embed
+	mux     sync.Mutex
+	log     *util.Logger
+	apikey  string
+	domain  string
+	data    api.Rates
+	updated time.Time
+}
+
+var _ api.Tariff = (*Entsoe)(nil)
+
+func init() {
+	registry.Add("entsoe", NewEntsoeFromConfig)
+}
+
+func NewEntsoeFromConfig(other map[string]interface{}) (api.Tariff, error) {
+	var cc struct {
+		embed         `mapstructure:",squash"`
+		Securitytoken string
```

Comment:
Could we make this the usual "token"?

Reply:
Concurred, 5489f79d7dcd8155080edb3c0c413fb867893a98